### PR TITLE
feat(rpc-types-anvil): add `Index`, fix compatibility

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Closes: https://github.com/alloy-rs/alloy/issues/838

Required for: https://github.com/foundry-rs/foundry/issues/8084

Related PR in Foundry: https://github.com/foundry-rs/foundry/pull/8186

## Solution

Adds `Index`, previously missing from the `rpc-types-anvil`

Fixes deserialization issue in `MineOptions`

Also added some tests for `Forking`, `Index` and `MineOptions` serialization as this was giving trouble in Foundry 

## PR Checklist

- [x] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Comment by zerosnacks** [comment]
Moved back to draft as some more compatibility fixes may be required

File: crates/rpc-types-anvil/src/lib.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ impl<'de> serde::Deserialize<'de> for Forking {
         #[serde(rename_all = "camelCase")]
         struct ForkOpts {
             json_rpc_url: Option<String>,
-            #[serde(default, with = "alloy_serde::quantity::opt")]
+            #[serde(default, with = "alloy_serde::num::u64_opt_via_ruint")]
```
Comment (DaniPopes) [inline crates/rpc-types-anvil/src/lib.rs:34]: please use quantity

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Comment by zerosnacks** [comment]
Compatibility fixes have been applied, see passing CI in https://github.com/foundry-rs/foundry/pull/8186

**Review by mattsse — APPROVED** [review]

**Review by onbjerg — CHANGES_REQUESTED** [review]
pls update the changelog as well 🥇

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**